codegen-server-test/model/pokemon.smithy

```diff
@@ -0,0 +1,79 @@
+$version: "1.0"
+
+namespace com.aws.example
+
+use aws.protocols#restJson1
+
+/// The Pokémon Service allows you to retrieve information about Pokémon species.
+@title("Pokémon Service")
+@restJson1
+service PokemonService {
+    version: "2021-12-01",
+    resources: [PokemonSpecies],
+    operations: [GetServerStatistics],
+}
+
+/// A Pokémon species forms the basis for at least one Pokémon.
+@title("Pokémon Species")
+resource PokemonSpecies {
+    identifiers: {
+        name: String
+    },
+    read: GetPokemonSpecies,
+}
+
+/// Retrieve information about a Pokémon species.
+@readonly
+@http(uri: "/pokemon-species/{name}", method: "GET")
+operation GetPokemonSpecies {
+    input: GetPokemonInput,
+    output: GetPokemonOutput,
+    errors: [ResourceNotFoundException],
+}
+
+structure GetPokemonInput {
+    @required
+    @httpLabel
+    name: String
+}
+
+structure GetPokemonOutput {
+    /// The name for this resource.
+    name: String,
+
+    /// A list of flavor text entries for this Pokémon species.
+    flavorTextEntries: FlavorTextEntries
+}
+
+/// Retrieve HTTP server statistiscs, such as calls count.
+@readonly
+@http(uri: "/stats", method: "GET")
+operation GetServerStatistics {
+    input: GetServerStatisticsInput,
+    output: GetServerStatisticsOutput,
+}
+
+structure GetServerStatisticsInput { }
```

Comment:
I've learnt a bunch of best practices about modelling in Smithy after I wrote this. Let's add `@input` and `@output` to all operation inputs / outputs. https://awslabs.github.io/smithy/1.0/spec/core/type-refinement-traits.html#smithy-api-input-trait